packages/eslint-plugin/src/rules/restrict-plus-operands.ts

```diff
@@ -1,20 +1,21 @@
 import type { TSESTree } from '@typescript-eslint/utils';
+import * as tsutils from 'tsutils';
 import * as ts from 'typescript';
 
 import * as util from '../util';
 
 type Options = [
   {
-    checkCompoundAssignments?: boolean;
     allowAny?: boolean;
+    allowBoolean?: boolean;
+    allowNullish?: boolean;
```

Comment:
[The nullish case will create TS errors for numbers, and is only valid in strings](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEYD2A7AzgF3gDwFzxQFcBbAIxBngB8DCIIBuAWAChXt4BqeARmZY7ds-QfADkY-q1CRYCZOiwBPfETIVq8QilAAzAJYoQwKSyVde-c9yVWLE06NusgA).
In the string case it also means you'll be concatenating with `'null' | 'undefined'` - which doesn't seem desirable in the vast majority of code.

I don't think that this option really makes sense in the context of the rule.